capstone/src/test/java/com/google/sps/RecommenderAlgorithmTest.java

```diff
@@ -0,0 +1,422 @@
+package com.google.sps;
+ 
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+ 
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.Room;
+import com.google.sps.data.User;
+import com.google.sps.data.Tag;
+import com.google.sps.data.RecommenderAlgorithm;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Key;
+import com.google.sps.service.DatabaseService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+ 
+@RunWith(JUnit4.class)
+public final class RecommenderAlgorithmTest {
```

Comment:
None of these tests recommend the most common scenario in which there are more people and rooms than need to be recommended, and its able to reccomend the best ones. All of them end up asserting that every room in the test gets recommended, not that the best ones do. Add at least one test which selects the best recommendation out of several options (probably done by lowering the recommendation limit to like 2, and verify that some were selected over others)